twilight-validate/src/component/component_v2.rs

```diff
@@ -0,0 +1,181 @@
+//! Validates components V2.
+
+use super::{ComponentValidationError, ComponentValidationErrorType};
+use twilight_model::channel::message::component::{
+    Container, MediaGallery, MediaGalleryItem, Section, TextDisplay, Thumbnail,
+};
+use twilight_model::channel::message::Component;
+
+/// Maximum length of text display content.
+pub const TEXT_DISPLAY_CONTENT_LENGTH_MAX: usize = 2000;
+
+/// Minimum amount of items in a media gallery.
+pub const MEDIA_GALLERY_ITEMS_MIN: usize = 1;
+
+/// Maximum amount of items in a media gallery.
+pub const MEDIA_GALLERY_ITEMS_MAX: usize = 10;
+
+/// Maximum length of a description of a media gallery item.
+pub const MEDIA_GALLERY_ITEM_DESCRIPTION_LENGTH_MAX: usize = 1024;
+
+/// Minimum amount of components in a section.
+pub const SECTION_COMPONENTS_MIN: usize = 1;
+
+/// Maximum amount of components in a section.
+pub const SECTION_COMPONENTS_MAX: usize = 3;
+
+/// Maximum length of a thumbnail description
+pub const THUMBNAIL_DESCRIPTION_LENGTH_MAX: usize = 1024;
+
+// TODO: rewrite comment
```

Comment:
Then please rewrite it :) Errors section could probably link to the individual methods, and those definitely also need an errors section.

Also, `todo!()` is a no-go for me, please handle all branches and remove it

Reply:
I have pushed a commit that resolves this